### DISCUSSION TITLE
Mac mini HDMI port - my monitor does not work as expected

### DISCUSSION THREAD

**Discussion by waydabber** [MonitorControl/MonitorControl#599] [Q&A] [answered]
My display connected to the HDMI port of an M1 or 2018 Intel mac mini does not work! Can you help me?

**Comment by waydabber** [comment, marked as answer]
Hardware DDC control through the HDMI port of the 2018 Intel and the 2020 M1 mini won't work as DDC communication is blocked on these ports. There is no known way to get around this limitation. You can however use software mode on such a display to control brightness using MonitorControl!

We recommend using USB-C on the 2018 and M1 mini.

**But why DDC is it blocked on these ports?**

There is probably no real valid reason behind it, it might simply be a bug either in hardware or software design. Since Apple does not regard DDC communication as a supported feature, they don't care and offer no support regarding this problem. :(

**Will this be fixed in the future?**

It might be fixed. Apple may issue a firmware update or change something in macOS to allow communication. Alternatively it might be that we simply did not figure out how to do it, maybe there is a simple fix yet to be discovered. Control through existing ports was discovered via trial&error and sometimes sheer luck, so anything can happen.

**Is there a way around this limitation? Can I use a HDMI-DisplayPort dongle for example?**

HDMI-DisplayPort dongles do not seem to work to fix this issue. :(

**I still want to tinker with it, maybe it works for me!**

MonitorControl currently actively disables the HDMI port of the M1 mini as an attempt to control it made some displays behave erratically (this might be an indication that something is getting through). Control is not actively disabled for 2018 Mini - it just doesn't work. If you want to force-enable control on the M1 mini, you can do this by changing a few lines and recompiling the app from source. 

You can simply remove lines 275-277 here and build the app in XCode by following the build instructions.

https://github.com/MonitorControl/MonitorControl/blob/6bba2f6ed749afd0b7da32c0769d01bf9a4ef25e/MonitorControl/Support/Arm64DDC.swift#L275

**Reply by jdz321** [reply]
🤔 Can I use type-c to DisplayPort adaptor?

**Reply by waydabber** [reply]
Yes.

(but please avoid USB [DisplayLink](https://www.synaptics.com/products/displaylink-graphics) adapters as they won't work.)
